#!/usr/bin/env python
PACKAGE = "franka_human_friendly_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness_X", double_t, 0, "Cartesian translational stiffness", 400, 0, 4000)
gen.add("translational_stiffness_Y", double_t, 0, "Cartesian translational stiffness", 400, 0, 4000)
gen.add("translational_stiffness_Z", double_t, 0, "Cartesian translational stiffness", 400, 0, 4000)
gen.add("rotational_stiffness_X", double_t, 0, "Cartesian rotational stiffness", 30, 0, 50)
gen.add("rotational_stiffness_Y", double_t, 0, "Cartesian rotational stiffness", 30, 0, 50)
gen.add("rotational_stiffness_Z", double_t, 0, "Cartesian rotational stiffness", 30, 0, 50)

#Use this if you want to use rotation matrix to change the orientation of the stiffness matrix
gen.add("roll", double_t, 0, "stiffness frame rotation roll", 0,0, 3.14159265359)
gen.add("pitch", double_t, 0, "stiffness frame rotation pitch", 0,0, 3.14159265359)
gen.add("yaw", double_t, 0, "stiffness frame rotation yaw", 0, 0, 3.14159265359)

gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0, 0, 100)

gen.add("max_delta_lin", double_t, 0, "Maximum delta linear displacement [m]", 0.05, 0.01, 0.1)

gen.add("max_delta_ori", double_t, 0, "Maximum delta angular displacement [rad]", 0.1, 0.01, 0.3)

exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param"))
